권한 계층 적용

10kseok opened the pull request:
권한 계층을 메서드 보안에서도 사용할 수 있게 하였습니다.

10kseok committed:
feat: 🎸 수정 인증 헤더 인터셉터 구현

#86  restTemplate 동작시 인증 헤더를 붙여주는 인터셉터를 구현하였습니다. 이제는 따로 header를 주입받지 않고 사용해도 API 호출이 가능합니다.

10kseok committed:
feat: 🎸 수정 인증 헤더 인터셉터 구현

#86  restTemplate 동작시 인증 헤더를 붙여주는 인터셉터를 구현하였습니다. 이제는 따로 header를 주입받지 않고 사용해도 API 호출이 가능합니다.

10kseok committed:
Merge branch 'feature/add-authorization-header' of https://github.com/nhnacademy-be3-CookShoong/cookshoong-frontend into feature/add-authorization-header

10kseok committed:
refactor: 💡 수정 시큐리티 필터무시 url 분리

시큐리티에 아예 제외시킬 url은 static에 담긴 파일로만 제한을 두게끔하고, 기존에 포함되어있던 url은 추후 바뀔 수도 있는 것을 고려하여 permitAll을 통해 관리하게끔 하였습니다.

10kseok committed:
feat: 🎸 추가 권한계층을 메서드 보안에도 적용

권한 계층을 설정해둔 것이 메서드 보안에는 적용되어있지 않아서
메서드 보안(@PreAuthorize, @PostAuthorize, ...)에 적용되게끔 핸들러를 추가해주었습니다.

10kseok committed:
refactor: 💡 수정 권한별 뷰 설정

권한 계층이 적용되면서 겹치게 보이던 이슈(상위 권한이 하위 권한뷰를 포함하여 노출)가 있었습니다.
각 권한에 맞게 보여주기 위해 시큐리티 태그를 사용하여 수정 작업을 진행하였습니다.

10kseok committed:
feat: 🎸 추가 권한 어댑터 구현

인증 객체를 생성할 때 인증객체 내부에서 액세스 토큰을 파싱해서 권한을 얻어냈었는데, 인증 객체는 이 과정을 알 필요가 없다고 판단하여
어댑터를 통해 권한 객체를 주입받도록 하였습니다.

10kseok committed:
refactor: 💡 수정 RestTemplate Readtime 변경

디버깅으로 할 때 자주 Readtimeout이 일어나 시간을 10초로 늘려주었습니다.

unhas01 commented on src/main/java/store/cookshoong/www/cookshoongfrontend/auth/adapter/JwtRoleAdapter.java:
```diff
@@ -0,0 +1,24 @@
+        return List.of(new SimpleGrantedAuthority(PREFIX + JwtResolver.resolveToken(token, ParsedAccessToken.class)
+            .getAuthority()));
+    }
+}
```
기본생성자로 `private`으로 막은것으로 보아 유틸 클래스로 보입니다.
클래스이름에 Util이 들어가면 보기가 좋을 것 같은데 `JwtRoleUtil` 같은 이름의 클래스는 어떤가요?

추가로 Java doc 없습니다.

10kseok replied:
다시 생각해보니 같은 기능과 의미를 가지는 클래스(JwtResolver)가 있어서 해당 클래스 내부로 넣어줬습니다!

unhas01 commented on src/main/java/store/cookshoong/www/cookshoongfrontend/auth/interceptor/RestTemplateHeaderModifierInterceptor.java:
```diff
@@ -0,0 +1,35 @@
+ * @author koesnam (추만석)
+ * @since 2023.07.24
+ */
+public class RestTemplateHeaderModifierInterceptor implements ClientHttpRequestInterceptor {
```
메서드에 java doc없습니다

10kseok replied:
오버라이드한 메서드라 빼줬지만 이해를 돕기위해 추가했습니다!

unhas01 commented on src/main/java/store/cookshoong/www/cookshoongfrontend/auth/interceptor/RestTemplateHeaderModifierInterceptor.java:
```diff
@@ -0,0 +1,35 @@
+    @Override
+    public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttpRequestExecution execution)
+        throws IOException {
+        SecurityContext session = SecurityContextHolder.getContext();
```
`SecurityContext` 변수명으로 session은 뭔가 이질감이 있어 보입니다..

10kseok replied:
context로 변경하였습니다~~

unhas01 reviewed (commented):
수고하셨습니다.

10kseok committed:
refactor: 💡 수정 JwtRoleAdapter 클래스 JwtResolver로 이동

JwtRoleAdapter가 JwtResolver와 기능이 동일한 것 같아 JwtRoleAdapter를 없애고 JwtResolver 내부에 메서드로 추가하였습니다.

10kseok committed:
style: 💄 수정 SecurityContext 변수명 변경

SecurityContext를 가져온 변수명을 좀 더 가독성있게 변경하였습니다.

10kseok committed:
style: 💄 수정 MainViewController에 공백 추가

누락됐던 공백을 추가해주었습니다.

10kseok committed:
docs: ✏️ 추가 JavaDoc 작성

이해를 돕기 위해 자바독을 추가해주었습니다!

JeongJeWan reviewed (approved):
확인했습니다@.@
